ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/damle/SandboxDamle.scala

```diff
@@ -3,6 +3,8 @@
 
 package com.digitalasset.platform.sandbox.damle
 
+import java.util.concurrent.ConcurrentHashMap
```

Comment:
You may want to consider Scala's [`TrieMap`](https://www.scala-lang.org/api/2.12.2/scala/collection/concurrent/TrieMap.html) for this.

Reply:
@gabor-aranyossy has some stories from the trenches to tell about `TrieMap`. For that reason I'm not too unhappy to see `ConcurentHashMap` being used.